Mathlib/Algebra/GroupWithZero/Basic.lean

```diff
@@ -149,7 +149,7 @@ lemma zero_pow_eq (n : ℕ) : (0 : M₀) ^ n = if n = 0 then 1 else 0 := by
 lemma zero_pow_eq_one₀ [Nontrivial M₀] : (0 : M₀) ^ n = 1 ↔ n = 0 := by
   rw [zero_pow_eq, one_ne_zero.ite_eq_left_iff]
 
-lemma pow_eq_zero_of_le : ∀ {m n} (_ : m ≤ n) (_ : a ^ m = 0), a ^ n = 0
+lemma pow_eq_zero_of_le : ∀ {m n}, m ≤ n → a ^ m = 0 → a ^ n = 0
```

Comment:
These changes are an improvement already. I wonder: should you make `m` and `n` arguments and move the two hypotheses left of the colon, while you're at it?

Reply:
I don't think the proof by induction (by pattern matching) still work if you do that.